client/src/featureform/resources.py

```diff
@@ -2278,9 +2280,27 @@ def __init__(
                 "The DatabricksCredentials requires only one credentials set ('username' and 'password' or 'host' and 'token' set.)"
             )
 
-        if not cluster_id:
+        if not self.cluster_id:
             raise Exception("Cluster_id of existing cluster must be provided")
 
+        if not self._validate_cluster_id():
+            raise ValueError(
+                "Cluster_id must follow the format: xxx-xxxxxx-xxxxxxxx, where x is alphanumeric"
```

Comment:
Lets change this to something like:
`f"Invalid cluster_id: expected id in the format 'xxx-xxxxxx-xxxxxxxx' but received '{self.cluster_id}'`

Reply:
Changed here - 246fc8d0d2b2949838806b0b795b04f69daba214